scipy/optimize/_minpack_py.py

```diff
@@ -639,6 +641,18 @@ def curve_fit(f, xdata, ydata, p0=None, sigma=None, absolute_sigma=False,
         `mesg`, and `ier`.
 
         .. versionadded:: 1.9
+    nan_policy : {'propagate', 'raise', 'omit', None}, optional
+        Defines how to handle when input contains nan.
+        The following options are available (default is 'None'):
+
+          * 'propagate': just propagate nan values without check finite.
+          * 'raise': throws an error
+          * 'omit': performs the calculations ignoring nan values without check finite.
+          * None: check nan valudes based on `check_finite` value.
+
+        Note that if thsi value is specified explicitly (not None), `check_finite` will be set as False.
+
+        .. versionadded:: 1.10
```

Comment:
I'm not sure this PR is too late for 1.10?

Reply:
Overally this will probably be bumped to 1.11.